src/main/java/org/folio/util/OkapiUtil.java

```diff
@@ -11,27 +11,53 @@
 import io.vertx.core.Future;
 import io.vertx.core.http.HttpMethod;
 import io.vertx.core.json.JsonArray;
+import org.folio.okapi.common.ModuleId;
+import org.folio.okapi.common.XOkapiHeaders;
 import org.folio.rest.tools.utils.TenantTool;
 
 public final class OkapiUtil {
 
   private OkapiUtil() {
   }
 
-  public static Future<List<String>> getModulesProvidingInterface(String interfaceName, Map<String, String> okapiHeaders) {
+  /**
+   * Set the module id header for the interface.
+   *
+   * <p>See <a href="https://github.com/folio-org/okapi/blob/master/doc/guide.md#multiple-interfaces">
+   * multiple interfaces</a> documentation.
+   *
+   * @param interfaceName the interface with "interfaceType": "multiple", for example custom-fields
+   * @param moduleName the module name (product name) without version, for example mod-users
+   * @param okapiHeaders where to get the "X-Okapi-Tenant" header and where to write the "X-Okapi-Module-Id" header
+   * @return
+   */
+  public static Future<Void> setModuleIdForMultipleInterface(
+      String interfaceName, String moduleName, Map<String, String> okapiHeaders) {
 
     String requestUri = String.format(GET_MODULES_WITH_INTERFACE, TenantTool.tenantId(okapiHeaders), interfaceName);
     return HttpClientUtil.getRequestOkapi(HttpMethod.GET, okapiHeaders, requestUri)
         .expect(SC_OK)
         .send()
-        .map(res -> extractModuleIds(res.bodyAsJsonArray()))
-        .recover(e -> HttpClientUtil.errorManagement(e, "Failed to get modules providing interface " + interfaceName));
+        .compose(res -> {
+          var list = extractModuleIds(res.bodyAsJsonArray(), moduleName);
+          if (list.isEmpty()) {
+            return Future.failedFuture("No module found.");
+          }
+          if (list.size() != 1) {
+            return Future.failedFuture("Multiple modules found: " + list.toString());
+          }
+          okapiHeaders.put(XOkapiHeaders.MODULE_ID, list.get(0));
```

Comment:
We're setting the `X-Okapi-Module-Id` here based on the `moduleName` provided as a parameter. But isn't it the callers job to set this header? Or is it that what is calling this method knows which interface to call for the custom fields?

Say we have two modules implementing the same custom fields interface. The user import service knows which one to call, and that's what this method is intended to handle. Do I have that right?

Reply:
I see it's because we have multiple modules (two) that now provide the custom fields interface (mod orders storage and mod users). This module (the caller) needs to specify which one for okapi to call.